File: easy/protein-translation.js
const CODONS = {
  AUG: "Methionine",
  UUC: "Phenylalanine",
  UUU: "Phenylalanine",
  UUA: "Leucine",
  UUG: "Leucine",
  UCU: "Serine",
  UCC: "Serine",
  UCA: "Serine",
  UCG: "Serine",
  UAU: "Tyrosine",
  UAC: "Tyrosine",
  UGU: "Cysteine",
  UGC: "Cysteine",
  UGG: "Tryptophan",
  UAA: "STOP",
  UAG: "STOP",
  UGA: "STOP",
};

export const translate = (RNA = "") => {
  const PROTEIN = [];
  for (let i = 0; i < RNA.length; i += 3) {
    if (CODONS[RNA.slice(i, i + 3)] === "STOP") {
      break;
    }
    if (!CODONS[RNA.slice(i, i + 3)]) {
      throw new Error("Invalid codon");
    }
    PROTEIN.push(CODONS[RNA.slice(i, i + 3)]);
  }
  return PROTEIN;
};

File: python/protein_translation.py
def proteins(strand):
    protein = []
    for i in range(len(strand) // 3):
        codon = strand[3 * i:3 * i + 3]
        if codon in ["AUG"]:
            protein.append("Methionine")
        elif codon in ["UUU", "UUC"]:
            protein.append("Phenylalanine")
        elif codon in ["UUA", "UUG"]:
            protein.append("Leucine")
        elif codon in ["UCU", "UCC", "UCA", "UCG"]:
            protein.append("Serine")
        elif codon in ["UAU", "UAC"]:
            protein.append("Tyrosine")
        elif codon in ["UGU", "UGC"]:
            protein.append("Cysteine")
        elif codon in ["UGG"]:
            protein.append("Tryptophan")
        else:
            break
    return protein
